featurebyte/config.py

```diff
@@ -357,25 +390,28 @@ def _parse_config(self, path: Path) -> None:
         if logging_settings:
             # parse logging settings
             self.logging = LoggingSettings(**logging_settings)
+        else:
+            self.logging = LoggingSettings()
 
         storage_settings = self.settings.pop("storage", None)
         if storage_settings:
             # parse storage settings
             self.storage = LocalStorageSettings(**storage_settings)
+        else:
+            self.storage = LocalStorageSettings()
 
         profile_settings = self.settings.pop("profile", None)
         if profile_settings:
             # parse profile settings
             self.profiles = ProfileList(profiles=profile_settings).profiles
 
-        if self.profiles:
+            # Set default _profile if specified
             profile_map = {profile.name: profile for profile in self.profiles}
             default_profile = self.settings.pop("default_profile", None)
-            selected_profile_name = os.environ.get("FEATUREBYTE_PROFILE")
-            if selected_profile_name:
-                self._profile = profile_map.get(selected_profile_name)
-            else:
-                self._profile = profile_map.get(default_profile)
+            self._profile = profile_map.get(default_profile)
+        else:
+            self.profiles = []
+            self._profile = None
```

Comment:
due to the new singleton, if config files are missing fields, have to force default values back, the getters will return the oldest valid value, which might not be present in the "active" config file

Reply:
could we reset the instance variables in the constructor as well, i.e reset them under the `force or config_file_path present` path? that way i think we should be able to skip all the `else` blocks here

Reply:
refactored